### PR TITLE
feat: Add support for bindParameters

### LINKED ISSUE #83: feat: Add support for dbstmt.bindParameters()
**Describe the solution you'd like**

As of idb-connector 1.2.7 [dbstmt.bindParameters](https://github.com/IBM/nodejs-idb-connector/tree/master/docs#dbstmtbindparametersparams-callback) API is available.

This API accepts an array of values rather than 2D array. 

We should add support for it within this project as well

### PR DESCRIPTION
**Pull request by abmusse**
Resolves #83

### CODE REVIEW

File: test/statementTest.js [hunk 1]
```diff
@@ -168,6 +168,44 @@ describe('Statement Class Tests', () => {
     });
   });
 
+  describe('bindParameters', () => {
+    before('create table for test', async () => {
+      const pool = new DBPool({ url: '*LOCAL' }, { incrementSize: 2 });
+      const createTable = `CREATE TABLE ${schema}.SCORES(team VARCHAR(100) ALLOCATE(20), score INTEGER)`;
```
Comment (abmusse) [inline test/statementTest.js:174]: In [#78](https://github.com/IBM/nodejs-idb-pconnector/pull/78/files#diff-860887daeecc0a57c5a03dcca9925ff1e1ffde296709a5600c2cba3c09a27c52R31) The test scores table is added. If that gets merged first no need to duplicate creating it here.

Comment (kadler) [reply test/statementTest.js]: That was merged. There are now conflicts which can be cleaned up and I'm guessing this can then be removed?

File: lib/statement.js [hunk 2]
```diff
@@ -64,9 +65,31 @@ class Statement {
    * @memberof Statement
    */
   async bind(params) {
+    deprecate('As of 1.1.0, bind() is deprecated use bindParameters() instead.');
```
Comment (kadler) [inline lib/statement.js:68]: FYI since bindParam also throws a deprecation warning, this would result in double warnings. Not sure if good or bad. Would have to duplicate bindParam here or separate the body of bindParam in to an internal function and make both wrappers on top of the new function.

**Review by kadler — APPROVED** [review]
Looks OK to me, pending the conflict resolution and the double deprecation warning.

File: test/statementTest.js (hunk 1 above)
Comment (abmusse) [reply test/statementTest.js]: Yes This can just be cleaned up by removing the `afterEach` hook.

File: lib/statement.js (hunk 2 above)
Comment (abmusse) [reply lib/statement.js:68]: Yes I will create a `bindParamWrapper()` that both `bindParam()` and `bind()` will call.

File: lib/statement.js [hunk 3]
```diff
@@ -57,14 +59,49 @@ class Statement {
...
+
   /**
    * Shorthand for bindParam
    * @param {Array} params - the parameter list
    * See `bindParam` for additional documentation
    * @memberof Statement
    */
   async bind(params) {
-    await this.bindParam(params);
+    deprecate('As of 1.1.0, bind() is deprecated use bindParameters() instead.');
```
Comment (abmusse) [inline lib/statement.js:82]: After running the tests the deprecation warning will be:

```
Statement deprecated As of 1.1.0, bind() is deprecated use bindParameters() instead. 
```

Perhaps we should reword this to:

```
Statement deprecated bind() as of 1.1.0. Use bindParameters() instead.
```

```suggestion
    deprecate('bind() as of 1.1.0. Use bindParameters() instead.');
```

File: lib/statement.js [hunk 4]
```diff
@@ -57,14 +59,49 @@ class Statement {
...
+  /**
+   * @description
+   * Associates parameter markers in an SQL statement to app variables
+   * @param {Array} params - the parameter list
+   * Each parameter element will also be an Array with 3 values (Value, In/out Type, Indicator)
+   * @returns {Promise} - Promise object represents the execution of bindParam()
+   * @memberof Statement
+   */
+  async bindParam(params) {
+    deprecate('As of 1.1.0, bindParam() is deprecated use bindParameters() instead.');
```
Comment (abmusse) [inline lib/statement.js:71]: After running the tests the deprecation warning will be:

```
Statement deprecated As of 1.1.0, bindParam() is deprecated use bindParameters() instead.
```

Perhaps we should reword this to:

```
Statement deprecated bindParam() as of 1.1.0. Use bindParameters() instead.
```

```suggestion
    deprecate('bindParam() as of 1.1.0. Use bindParameters() instead.');
```

File: lib/statement.js (hunk 3 above)
Comment (kadler) [reply lib/statement.js]: The added prefix is a bit weird, but I feel like the first one still flows better. If it read "Statement bind() deprecated as of 1.1.0. Use bindParameters() instead.", I'd agree though.

From a grammatical point of view, there does need to be something between the two clauses, though. "As of 1.1.0, bind() is deprecated — use bindParameters() instead." would work, "As of 1.1.0, bind() is deprecated and you should use bindParameters() instead." would also work, though less terse.

Comment (abmusse) [reply lib/statement.js]: OK I will go back with `As of 1.1.0, bind() is deprecated and you should use bindParameters() instead`
